review/get_review_ids.py

```diff
@@ -89,19 +94,43 @@ def get_vertical(current_course):
     vertical_data = set()
 
     for block_key, state in get_records(user, current_course):
-        if is_valid_problem(store, block_key, state):
-            parent = course_blocks.get_parents(block_key)[0]
-            vertical_id = parent.block_id
-            vertical_data.add(vertical_id)
-            delete_state_of_review_problem(user, current_course, block_key.block_id)
+        block_key = block_key.replace(course_key=store.fill_in_run(block_key.course_key))
+        if is_valid_problem(store, block_key, state, course_blocks):
+            # If the block_key does not have a subsection (sequential) in it's tree,
+            # we should skip it.
+            subsection = course_blocks.get_transformer_block_field(
+                block_key,
+                GradesTransformer,
+                'subsections',
+                set(),
+            )
+            if subsection:
+                vertical = course_blocks.get_parents(block_key)[0]
+                sequential = course_blocks.get_parents(vertical)[0]
+                # This is in case the direct parent of a problem is not a vertical,
+                # we want to keep looking until we find the parent vertical to display.
+                # For example, you may see:
+                # sequential -> vertical -> split_test -> problem
+                # OR
+                # sequential -> vertical -> vertical -> problem
+                # OR
+                # sequential -> vertical -> conditional_block -> problem
+                # Note: we know this will not be an infinite loop because with the
+                # above if statement, we are guaranteed there is a 'sequential' in
+                # the block's parent line.
+                while sequential.block_type != 'sequential':
+                    vertical = sequential
+                    sequential = course_blocks.get_parents(vertical)[0]
```

Comment:
Put L108-L123 within a `try`-`except` block to catch `IndexError`s.  This should be the simplest way to handle varying course structures (including blocks with multiple parents).